source/common/router/upstream_request.cc

```diff
@@ -371,6 +371,9 @@ void UpstreamRequest::onPoolReady(
   ENVOY_STREAM_LOG(debug, "pool ready", *parent_.callbacks());
   upstream_ = std::move(upstream);
 
+  // Have the upstream use the account of the downstream.
+  upstream_->setAccount(parent_.callbacks()->account());
```

Comment:
I had been wondering if this information could be available at the time we call newStreamEncoder from HttpConnPoolImplBase::onPoolReady via the HTTP context provided there.  I guess setting the account at this level is fine, it's hard to navigate all the levels of callbacks.